pd/src/components/ibc.rs

```diff
@@ -32,11 +50,98 @@ impl Component for IBCComponent {
         Ok(())
     }
 
-    async fn execute_tx(&mut self, _tx: &Transaction) -> Result<()> {
-        Ok(())
+    async fn execute_tx(&mut self, tx: &Transaction) -> Result<()> {
+        // handle client transactions
+        for action in tx.transaction_body.actions.iter() {
+            let ibc_action = match action {
+                Action::IBCAction(ibc_action) => Some(ibc_action),
+                _ => None,
+            };
+            if ibc_action.is_none() {
+                continue;
+            }
```

Comment:
The control flow here could be cleaned up with `filter_map`, by doing something like
```rust
        for action in tx.transaction_body.actions.iter()
            .filter_map(|action| match action {
                Action::IBCAction(ibc_action) => Some(ibc_action),
                _ => None,
            })
        {
            // ...
        }
```
so that there's no need for further filtering or unwrapping.